### ISSUE TITLE
Model/provider duplication and default provider not preserved when using export feature

### ISSUE THREAD

**Issue by twilsonco** [BoltAI/BoltAI#195]
**Describe the bug**
The BoltAI settings export (using "File -> Import -> From BoltAI JSON export; from #122) has a bug, such that all models/providers with non-default settings are *added* to the list of models, rather than replacing the blank defaults with the non-default values. This results in duplicates in the list of models/providers as in the below image (I hadn't noticed what was happening until I did the import twice, hence there being multiple copies of some providers in my screenshot)

Note that I also have an API setup for OpenAI, however *it was not duplicated in the list*. I'm not sure if this is because it's the default, or because it has a special implementation in BoltAI because it was the original provider supported, unique from the others.

*As a secondary issue, I had set Google AI as my default in the BoltAI instance I exported the JSON from, but that was not preserved when importing the exported data*.

**To Reproduce**
1. Setup one or more model/provider with an API key *that is not OpenAI*, and save changes
	* *also set one of the modified models/providers to be the default*
3. Use "File -> Export..." to export settings to a JSON file
4. Import that JSON file back into (a fresh installation of) BoltAI

**Expected behavior**
The list of models/providers should look identical to when the exported JSON was created, including the changed default model/provider.

**Actual behavior**
The models/providers for which an API key has been added will be found at the bottom of the list as duplicates, and their blank counterparts will remain higher in the list where they are in a fresh BoltAI installation, and OpenAI will remain the default model/provider.

**Screenshots**
![CleanShot 2024-10-25 at 23 33 30](https://github.com/user-attachments/assets/f23a87fc-6416-4c71-993c-7620a3fc2f3d)

**Version (please complete the following information):**
 - macOS: 15.0.1
 - app version 1.25.0(114) (non-SetApp)

**Comment by longseespace** [comment]
Thanks for the report. I'll investigate further later today.

**Comment by twilsonco** [comment]
Thanks. I should clarify that it's not only OpenAI that wasn't duplicated. I also had an API key set for Replicate, and that wasn't duplicated. So I'm not sure what determines which are duplicated. I assume the providers that were added later were done so using some abstraction you devised to make it easier to add new providers, and that those are the ones affected.

**Comment by longseespace** [comment]
You’re correct. In the early version, BoltAI only supports one configuration for OpenAI and Replicate. And those API keys are stored in the Keychain directly.

Other providers are stored in the SQLite database.

This is not necessarily a bug I think. But it’s indeed an UX issue.

Currently each configuration doesn’t have a global unique ID. And so when you import it, it will always insert new records into the db and not replacing old records (blank or not).

I think one way to fix this is to add a “import_uuid” to id each record. And maybe give the option to delete/replace blank records. What do you think?

**Comment by twilsonco** [comment]
That makes sense, as does the proposed solution. I also agree it's a UX issue and not a bug.

To avoid a breaking change, the revised import code should include the ability to process exported JSON files from before this change was made, in which case the current behavior (duplicated provider/model) would be expected.
